storage/imxrt-flash/rom_api.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT ROM API driver for FlexSPI
+ *
+ * Copyright 2019 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _FLEXSPI_ROM_API_H_
+#define _FLEXSPI_ROM_API_H_
+
+#include <stdbool.h>
+#include <stdio.h>
+
+typedef struct _serial_nor_config_option_t {
+	 union {
+		  struct {
+			   uint32_t max_freq : 4;			// Maximum supported Frequency
+			   uint32_t misc_mode : 4;			// miscellaneous mode
+			   uint32_t quad_mode_setting : 4;	// Quad mode setting
+			   uint32_t cmd_pads : 4;			// Command pads
+			   uint32_t query_pads : 4;			// SFDP read pads
+			   uint32_t device_type : 4;		// Device type
+			   uint32_t option_size : 4;		// Option size, in terms of uint32_t, size = (option_size + 1) * 4
+			   uint32_t tag : 4;				// Tag, must be 0x0E
+		  } B;
+		  uint32_t U;
+	 } option0;
+	 union {
+		  struct {
+			   uint32_t dummy_cycles : 8;		// Dummy cycles before read
+			   uint32_t reserved0 : 8;			// Reserved for future use
+			   uint32_t pinmux_group : 4;		// The pinmux group selection
+			   uint32_t reserved1 : 8;			// Reserved for future use
+			   uint32_t flash_connection : 4;	// Flash connection option: 0 - Single Flash connected to port A
+		  } B;
+		  uint32_t U;
+	 } option1;
+} serial_nor_config_option_t;
+
+
+typedef enum _flexspi_operation_t {
+	 kFlexSpiOperation_Command,					// FlexSPI operation: Only command, both TX and RX buffer are ignored.
+	 kFlexSpiOperation_Config,					// FlexSPI operation: Configure device mode, the TX FIFO size is fixed in LUT.
+	 kFlexSpiOperation_Write,					// FlexSPI operation: Write, only TX buffer is effective
+	 kFlexSpiOperation_Read,					// FlexSPI operation: Read, only Rx Buffer is effective.
+	 kFlexSpiOperation_End = kFlexSpiOperation_Read,
+} flexspi_operation_t;
+
+
+typedef struct _flexspi_xfer_t {
+	 flexspi_operation_t operation;				// FlexSPI operation
+	 uint32_t baseAddress;						// FlexSPI operation base address
+	 uint32_t seqId;							// Sequence Id
+	 uint32_t seqNum;							// Sequence Number
+	 bool isParallelModeEnable;					// Is a parallel transfer
+	 uint32_t *txBuffer;						// Tx buffer
+	 uint32_t txSize;							// Tx size in bytes
+	 uint32_t *rxBuffer;						// Rx buffer
+	 uint32_t rxSize;							// Rx size in bytes
+} flexspi_xfer_t;
```

Comment:
As above, align those comments using spaces and use C style comments. I would prefer that you don't use bool type, but I won't enforce it.